src/modules/file/file.controller.ts

```diff
@@ -369,4 +372,62 @@ export class FileController {
 
     return files;
   }
+
+  @Get('/meta')
+  async getFileMetaByPath(
+    @UserDecorator() user: User,
+    @Query('path') encodedPath: string,
+  ) {
+    const filePath = Buffer.from(encodedPath, 'base64').toString('utf-8');
+    if (!filePath || filePath.length === 0 || !filePath.includes('/')) {
+      throw new BadRequestException(
+        'Invalid path provided (it must be base64 encoded)',
+      );
+    }
+
+    try {
+      const path = getPathFileData(filePath);
+      const fileExt = path.fileType.length > 0 ? path.fileType : null;
+
+      const rootFolder = await this.folderUseCases.getFolderByUserId(
```

Comment:
There is business logic here, which should be part of a use case, probably called `getFileMetadataByPath`, otherwise, we are mixing controller and use case responsibilities.